src/components/v2v3/shared/FundingCycleConfigurationDrawers/NftDrawer/index.tsx

```diff
@@ -78,74 +115,121 @@ export default function NftDrawer({
   } = useFundingCycleDrawer(onClose)
 
   const onNftFormSaved = useCallback(async () => {
+    if (!rewardTiers) return
     setSubmitLoading(true)
-    const marketplaceFormValues = marketplaceForm.getFieldsValue(true)
-    const collectionName = marketplaceFormValues.collectionName
-
-    const [rewardTiersCIDs, nftCollectionMetadataCID] = await Promise.all([
-      uploadNftRewardsToIPFS(rewardTiers),
-      uploadNftCollectionMetadataToIPFS({
-        collectionName: marketplaceFormValues.collectionName?.length
-          ? marketplaceFormValues.collectionName
-          : defaultNftCollectionName(projectName),
-        collectionDescription: marketplaceFormValues.collectionDescription
-          ?.length
-          ? marketplaceFormValues.collectionDescription
-          : defaultNftCollectionDescription(projectName),
-        collectionLogoUri: logoUri,
-        collectionInfoUri: infoUri,
-      }),
-    ])
-
-    dispatch(editingV2ProjectActions.setNftRewardsName(collectionName))
-    dispatch(
-      editingV2ProjectActions.setNftRewardsSymbol(
-        marketplaceFormValues.collectionSymbol,
-      ),
-    )
-    dispatch(
-      editingV2ProjectActions.setNftRewardsCollectionDescription(
-        marketplaceFormValues.collectionDescription,
-      ),
-    )
-    dispatch(editingV2ProjectActions.setNftRewardTiers(rewardTiers))
-    dispatch(
-      editingV2ProjectActions.setNftRewardsCollectionMetadataCID(
-        nftCollectionMetadataCID,
-      ),
-    )
+    // call `JBTiered721DelegateProjectDeployer.reconfigureFundingCyclesOf` when a project without NFT configures a new NFT collection
+    if (!initialRewardTiers?.length) {
+      const marketplaceFormValues = marketplaceForm.getFieldsValue(true)
```

Comment:
**suggestion** can we break these out into multiple functions to improve readability?

Reply:
For sure